File: modules/apps/template/template-web/src/main/resources/META-INF/resources/js/modal/Field.js
/**
 * SPDX-FileCopyrightText: (c) 2000 Liferay, Inc. https://liferay.com
 * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
 */

import ClayForm from '@clayui/form';
import ClayIcon from '@clayui/icon';
import React from 'react';

export default function Field({children, errors, id, label, name}) {
	const errorMessage = errors[name];

	return (
		<ClayForm.Group className={errorMessage ? 'has-error' : null}>
			<label htmlFor={id}>
				{label}

				<span className="reference-mark">
					<ClayIcon symbol="asterisk" />
				</span>
			</label>

			{children}

			{errorMessage && (
				<div className="form-feedback-item">
					<span className="form-feedback-indicator mr-1">
						<ClayIcon symbol="exclamation-full" />
					</span>

					{errorMessage}
				</div>
			)}
		</ClayForm.Group>
	);
}

File: modules/apps/analytics/analytics-settings-web/src/main/java/com/liferay/analytics/settings/web/internal/model/Field.java
/**
 * SPDX-FileCopyrightText: (c) 2000 Liferay, Inc. https://liferay.com
 * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
 */

package com.liferay.analytics.settings.web.internal.model;

/**
 * @author Rachael Koestartyo
 */
public class Field {

	public Field(String category, String dataType, String name) {
		_category = category;
		_dataType = dataType;
		_name = name;
	}

	public String getCategory() {
		return _category;
	}

	public String getDataType() {
		return _dataType;
	}

	public String getName() {
		return _name;
	}

	public void setCategory(String category) {
		_category = category;
	}

	public void setDataType(String dataType) {
		_dataType = dataType;
	}

	public void setName(String name) {
		_name = name;
	}

	private String _category;
	private String _dataType;
	private String _name;

}